packages/ag-charts-community/src/chart/agChartV2.ts

```diff
@@ -507,10 +528,10 @@ function applyAxes(chart: Chart, options: { axes?: AgBaseAxisOptions[] }) {
 
 function applyLegend(chart: Chart, options: AgChartOptions) {
     const skip = ['listeners'];
-    chart.setLegendInit((legend) => {
-        applyOptionValues(legend, options.legend ?? {}, { skip });
-        if (options.legend?.listeners) {
-            Object.assign(chart.legend?.listeners!, options.legend.listeners ?? {});
+    chart.setLegendInit((legendKey, legend) => {
+        applyOptionValues(legend, (options as any)[legendKey] ?? {}, { skip });
+        if (chart.legend && options.legend?.listeners) {
+            Object.assign(chart.legend.listeners!, options.legend.listeners);
         }
     });
```

Comment:
Do we need to use a callback here? Since the distinct legend types are now on distinct property paths, can't they just be assigned options directly from `AgChartV2` rather than defering to later?